src/slashcommands/Update.ts

```diff
@@ -0,0 +1,80 @@
+import { Bot } from '../Bot';
+import {
+	Permissions,
+	CommandInteraction,
+	CacheType,
+	GuildMember,
+	TextChannel,
+	Guild,
+} from 'discord.js';
+import { SlashCommand } from './SlashCommand';
+import config from '../../config.json';
+import Enmap from 'enmap';
+
+export let updateChannels = new Enmap({ name: 'updateChannels' });
+
+export class Update implements SlashCommand {
+	public name = 'update';
+	public registerData = {
+		name: this.name,
+		description:
+			'Set the channel you wish to recieve Mirror update messages in',
+		options: [
+			{
+				name: 'channel',
+				description: 'The channel you wish to recieve update messages',
+				type: 7,
+				required: true,
+			},
+		],
+	};
+	public requiredPermissions = [Permissions.FLAGS.SEND_MESSAGES];
+
+	public async run(
+		bot: Bot,
+		interaction: CommandInteraction<CacheType>
+	): Promise<void> {
+		if (!(interaction.channel instanceof TextChannel)) {
+			interaction.reply('Command must be used in a server');
+			return;
+		}
+		let member = interaction.member as GuildMember;
+		if (
+			!member.permissionsIn(interaction.channel!).has('ADMINISTRATOR') ||
+			member.id != config.owner
+		) {
+			interaction.reply({
+				content:
+					'This command is only for people with Administrator permissions',
+				ephemeral: true,
+			});
+			return;
+		}
+		let channel = interaction.options.getChannel('channel');
+		if (
+			!interaction.guild?.me
```

Comment:
Why aren't we doing this check with the requiredPermissions array?

Reply:
No this is checking the new channel for permissions, assuming that maybe someone wants to assign mirror to a channel that they are currently not in